internal/status.go

```diff
@@ -0,0 +1,147 @@
+package internal
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/dustin/go-humanize"
+)
+
+type StatusLine struct {
+	resources []Resource
+
+	resourceSizes       []int64
+	totalBytes          int64
+	sizingSuccess       bool
+	bytesDownloaded     int64
+	resourcesDownloaded int
+	spinI               int
+	indexCh             chan int
+	startTime           time.Time
+}
+
+var spinChars = [5]string{"-", "\\", "|", "/", "-"}
+
+// Create, initialize, and start a new StatusLine.
+func NewStatusLine(resources []Resource) *StatusLine {
```

Comment:
The method should take a `ctx context.Context`. See https://pkg.go.dev/context for details.